src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -92,6 +92,28 @@ enum module_processing_mode {
 	MODULE_PROCESSING_BYPASS,
 };
 
+/**
+ * \struct input_stream_buffer
+ * \brief Input stream buffer
+ */
+struct input_stream_buffer {
+	void *data; /* data stream buffer */
+	uint32_t size; /* size of data in the buffer */
+	uint32_t consumed; /* number of bytes consumed by the module */
+
+	/* Indicates end of stream condition has occurred on the input stream */
+	bool end_of_stream;
+};
+
+/**
+ * \struct output_stream_buffer
+ * \brief Output stream buffer
+ */
+struct output_stream_buffer {
+	void *data; /* data stream buffer */
+	uint32_t size; /* size of data in the buffer */
+};
+
 /*****************************************************************************/
```

Comment:
I wonder if the naming is correct here or codec/generic.h is the right place for these structures. Do we plan to use this structures only for codec_adapter? If so then we should reflect that in naming.

Otherwise, maybe move the definitions in a generic place.

Reply:
@dbaluta codec_adapter itself will be replaced by module_adapter at some point soon. So these input and output buffers will eventually be applicable to add modules. I could add the module_ prefix to clarify it